### PR TITLE
agent: Document the configuration options

### PR DESCRIPTION
**Pull request by wainersm**
The information about the configuration options of the agent is spread in documents and is incomplete. Let's have it proper documented.

While here I am sending a small change to the config.rs file.

Fixes: #4015

### CODE REVIEW

File: src/agent/README.md [hunk 1]
```diff
@@ -84,6 +84,33 @@ To install the `protoc` command on a Fedora/CentOS/RHEL system:
 $ sudo dnf install -y protobuf-compiler
 ```
 
+## Configuring the agent
+
+The agent can be configured in a couple of ways.
+
+If running under the supervision of the Kata runtime, the agent will be automatically configured according to some properties in the runtime's `configuration.toml` file. For example, the `enable_debug` option under the `[agent.kata]` section can be used to set debug-level logs. However, not all the accepted configurations are mapped into `configuration.toml`, so any additional options should be passed directly to the agent via kernel parameters.
```
Comment (amshinde) [inline src/agent/README.md:91]: Would be good to point out how kernel parameters are passed.

Comment (liubin) [reply src/agent/README.md:91]: Agree with @amshinde , it's better to give an example like `kernel_params = "agent.log=debug agent.debug_console agent.debug_console_vport=1026 agent.hotplug_timeout=10"`.

Comment (wainersm) [reply src/agent/README.md:91]: Thanks for your comments @amshinde and @liubin ! I agree with your suggestion, I will send a v2 of this PR soon.

**Comment by wainersm** [comment]
Just update this PR to address the suggestions of @amshinde and @liubin

**Review by liubin — APPROVED** [review]

**Comment by jodh-intel** [comment]
I'm in two minds about this. Docs are good, but I'm not totally convinced we want to document these options. The official way to configure Kata is using the `configuration.toml` config file. The `kernel_params=` option is a necessary evil but users should not need to tweak most of the "behind the scenes" agent settings. If they did, we'd add a proper "knob" in the config file.

The other problem with documenting them is that we create a potential maintenance problem: it's harder to modify the behaviour of the agent settings (or even remove them).

An alternative to this PR would be to add them to the developer guide (to minimise the chance that real users (non developers) try to set these options and get into difficulties as a result).

If we **really** feel these need to be documented, I'd suggest a very clear warning message along the lines of:

> **Warning:**
>
> The official way to modify the Kata Containers agent behaviour is using the settings provided in the Kata Containers configuration file.
>
> The agent options listed here may change their behaviour between releases or they may even be removed (meaning setting them would silently fail).

The other problem with documenting the options of course is that the doc is not tied to the code so they are bound to deviate at some point.

I can see a possible compromise solution (not perfect, but maybe workable): define all the agent options in a YAML file that is parsed by the agent **at build time** and which is of course still human readable (we can add descriptions for all the options but the agnet wouldn't consume those).

**Comment by fidencio** [comment]
@wainersm, we're closing this PR as it's been opened and inactive for more than one year.
If you think this is a mistake and the PR should be re-opened, please, feel free to do so after rebasing it atop of current main.

**Pull request closed without merging**